Additional eventing sample edits (#2930)
knative/docs: docs/eventing/samples/helloworld/helloworld-python/README.md

  and a Docker Hub account configured (we'll use it for a container registry).

## Recreating the sample code

1. Create a new file named `helloworld.py` and paste the following code. This
   code creates a basic web server which listens on port 8080:

    ```python
    from flask import Flask, request, make_response
    import uuid

    app = Flask(__name__)

    @app.route('/', methods=['POST'])
    def hello_world():
        app.logger.warning(request.data)
        # Respond with another event (optional)
        response = make_response({
            "msg": "Hi from helloworld-python app!"
        })
        response.headers["Ce-Id"] = str(uuid.uuid4())
        response.headers["Ce-specversion"] = "0.3"
        response.headers["Ce-Source"] = "knative/eventing/samples/hello-world"
        response.headers["Ce-Type"] = "dev.knative.samples.hifromknative"
        return response

    if __name__ == '__main__':
        app.run(debug=True, host='0.0.0.0', port=8080)

    ```

1. Add a requirements.txt file containing the following contents:

    ```bash
    Flask==1.1.1
    ```
1. In your project directory, create a file named `Dockerfile` and copy the code

>    block below into it. For detailed instructions on dockerizing a Go app, see

Reviewer: Maybe we should replace the "dockerizing a go app" section with something related to Python, or possibly just remove the line related to Go?

        selector:
          app: helloworld-python
        ports:
        - protocol: TCP
          port: 80
          targetPort: 8080
    ---
    # Knative Eventing Trigger to trigger the helloworld-python service
    apiVersion: eventing.knative.dev/v1alpha1
    kind: Trigger
    metadata:
      name: helloworld-python
      namespace: knative-samples
    spec:
      broker: default
      filter:
        attributes:
          type: dev.knative.samples.helloworld
          source: dev.knative.samples/helloworldsource
      subscriber:
        ref:
          apiVersion: v1
          kind: Service
          name: helloworld-python
    ```

## Building and deploying the sample

Once you have recreated the sample code files (or used the files in the sample
folder) you're ready to build and deploy the sample app.

1. Use Docker to build the sample code into a container. To build and push with
   Docker Hub, run these commands replacing `{username}` with your Docker Hub
   username:

   ```shell
   # Build the container on your local machine
   docker build -t {username}/helloworld-python .

   # Push the container to docker registry
   docker push {username}/helloworld-python
   ```

> 1. After the build has completed and the container is pushed to docker hub, you

Reviewer: ```suggestion
1. After the build has completed and the container is pushed to Docker Hub, you
```

Updated for consistency.

   can deploy the sample application into your cluster. Ensure that the container image value
   in `sample-app.yaml` matches the container you built in the previous step. Apply
   the configuration using `kubectl`:

   ```shell
   kubectl apply --filename sample-app.yaml
   ```
    1.  Above command created a namespace `knative-samples` and labelled it with `knative-eventing-injection=enabled`, to enable eventing in the namespace. Verify using the following command:
        ```shell
        kubectl get ns knative-samples --show-labels
        ```
    1. It deployed the helloworld-python app as a K8s Deployment and created a K8s service names helloworld-python. Verify using the following command.
        ```shell
        kubectl --namespace knative-samples get deployments helloworld-python
        kubectl --namespace knative-samples get svc helloworld-python
        ```
    1. It created a Knative Eventing Trigger to route certain events to the helloworld-python application. Make sure that Ready=true
        ```shell
        kubectl --namespace knative-samples get trigger helloworld-python
        ```
## Send and verify CloudEvents

> Once you have deployed the application and verified that the namespace, sample application and trigger are ready, let's send a CloudEvent.

Reviewer: ```suggestion
Once you have deployed the application and verified that the namespace, sample application, and trigger are ready, let's send a CloudEvent.
```

Author: @mpetason I like what you have added for punctuation, but for consistency, 'let's' should not be used.

Better put would be - "Once you have deployed the application and verified that the namespace, sample application, and trigger are ready, a CloudEvent can be sent."

> Once you have deployed the application and verified that the namespace, sample application and trigger are ready, let's send a CloudEvent.

Reviewer 2: ```suggestion
After you have deployed the application, and have verified that the namespace, sample application and trigger are ready, you can send a CloudEvent.
```
Stick with the active voice and no fluff :) https://developers.google.com/style/voice

### Send CloudEvent to the Broker

> We can send an http request directly to the [Broker](../../../broker-trigger.md) with correct CloudEvent headers set.

Reviewer 2: ```suggestion
You can send an HTTP request directly to the Knative [broker](../../../broker-trigger.md) with correct CloudEvent headers set.
```

        {"msg":"Hi from Knative!"}

      ```

>   Play around with the CloudEvent attributes in the curl command and the trigger specification to understand how [Triggers work](../../../broker-trigger.md#trigger).

Reviewer 2: ```suggestion
  Try the CloudEvent attributes in the curl command and the trigger specification to understand how [triggers work](../../../broker-trigger.md#trigger).
```

## Verify reply from helloworld-python app

> `helloworld-python` app replies back with an event of `type= dev.knative.samples.hifromknative`, and `source=knative/eventing/samples/hello-world`. This event enters the eventing mesh via the Broker and can be delivered to other services using a Trigger

Reviewer 2: ```suggestion
The `helloworld-python` app replies with an event type `type= dev.knative.samples.hifromknative`, and source `source=knative/eventing/samples/hello-world`. The event enters the eventing mesh through the broker, and can be delivered to event sinks using a trigger
```
